crates/pathfinder/src/cairo/ext_py/service.rs

```diff
@@ -162,3 +178,32 @@ pub async fn start(
 
     Ok((handle, jh))
 }
+
+/// Returns if a new subprocess should be launched without wait
+fn on_joined_subprocess(
+    res: Result<Result<super::SubprocessExitInfo, anyhow::Error>, tokio::task::JoinError>,
+    processes_failed: &metrics::Counter,
+) -> bool {
+    match res {
+        Ok(Ok((pid, exit_status, exit_reason))) => {
+            info!(%pid, ?exit_status, ?exit_reason, "Subprocess exited");
+            let why = exit_reason.as_label();
+            metrics::increment_counter!(METRIC_EXITED_PROCESSES, "reason" => why);
```

Comment:
This is inconsistent with the other counter being accessed by ref.

Reply:
I guess I could add usizes for the reasons but ... is that better? metrics will have to handle this underneath already.